src/main/java/com/powsybl/openloadflow/network/impl/LfAreaImpl.java

```diff
@@ -0,0 +1,122 @@
+        this.interchangeTarget = area.getInterchangeTarget().orElse(0.0) / PerUnit.SB;
+        this.buses = buses;
+        this.boundaries = boundaries;
+    }
+
+    public static LfAreaImpl create(Area area, Set<LfBus> buses, Set<Boundary> boundaries, LfNetwork network, LfNetworkParameters parameters) {
+        LfAreaImpl lfArea = new LfAreaImpl(area, buses, boundaries, network, parameters);
+        lfArea.getBuses().forEach(bus -> bus.setArea(lfArea));
+        return lfArea;
+    }
+
+    private Area getArea() {
+        return areaRef.get();
+    }
+
+    @Override
+    public String getId() {
+        return getArea().getId();
+    }
+
+    @Override
+    public double getInterchangeTarget() {
+        return interchangeTarget;
+    }
+
+    @Override
+    public void setInterchangeTarget(double interchangeTarget) {
+        this.interchangeTarget = interchangeTarget;
+    }
+
+    @Override
+    public double getInterchange() {
+        return boundaries.stream().mapToDouble(Boundary::getP).sum();
+    }
+
+    @Override
+    public Set<LfBus> getBuses() {
+        return buses;
+    }
+
+    @Override
+    public Set<Boundary> getBoundaries() {
+        return boundaries;
+    }
+
+    @Override
+    public LfNetwork getNetwork() {
+        return network;
+    }
+
+    public static class BoundaryImpl implements Boundary {
+        private final LfBranch branch;
+        private final TwoSides side;
+
+        public BoundaryImpl(LfBranch branch, TwoSides side) {
+            this.branch = Objects.requireNonNull(branch);
+            this.side = Objects.requireNonNull(side);
+        }
+
+        @Override
+        public LfBranch getBranch() {
+            return branch;
+        }
+
+        @Override
+        public double getP() {
+            if (branch.isDisabled()) {
+                return 0.0;
+            }
+            if (branch instanceof LfTieLineBranch lfTieLineBranch) {
+                if (side == TwoSides.ONE) {
+                    DanglingLine danglingLine1 = lfTieLineBranch.getHalf1();
+                    double nominalV1 = danglingLine1.getTerminal().getVoltageLevel().getNominalV();
+                    return new SV(lfTieLineBranch.getP1().eval() * PerUnit.SB, lfTieLineBranch.getQ1().eval() * PerUnit.SB, lfTieLineBranch.getV1() * nominalV1, Math.toDegrees(lfTieLineBranch.getAngle1()), side)
+                            .otherSideP(danglingLine1, true) / PerUnit.SB;
+                } else if (side == TwoSides.TWO) {
+                    DanglingLine danglingLine = lfTieLineBranch.getHalf2();
+                    double nominalV2 = danglingLine.getTerminal().getVoltageLevel().getNominalV();
+                    return new SV(lfTieLineBranch.getP2().eval() * PerUnit.SB, lfTieLineBranch.getQ2().eval() * PerUnit.SB, lfTieLineBranch.getV2() * nominalV2, Math.toDegrees(lfTieLineBranch.getAngle2()), side)
+                            .otherSideP(danglingLine, true) / PerUnit.SB;
+                }
+            }
```

Comment:
splitShuntAdmittance to be set to false, see also https://github.com/powsybl/powsybl-core/pull/3169 and https://github.com/powsybl/powsybl-open-loadflow/pull/1098